public/js/charts/worstPMGraph.js

```diff
@@ -125,21 +132,34 @@ const PMTopFiveChart = async (type) => {
 
 async function handleLocationChange() {
   const nodes = await window.aq.getNodes();
-
-  const nodePMAverages = nodes.map((data) => {
+  const countyCitiesMap = await window.sheets.getCountyCitiesMap();
+  const allCountyNodes = Object.entries(countyCitiesMap)
+    .map(([countyName, countyCities]) => {
+      const countyNodes = nodes.filter(({ location }) =>
+        countyCities.some(
+          (city) =>
+            location && location.city && location.city.indexOf(city) >= 0
+        )
+      );
+      return { name: countyName, value: countyNodes };
+    })
+    // Don't show counties with 0 nodes
+    .filter(({ value }) => value.length !== 0)
+    .map((node) => node?.value);
+  const nodePMAverages = allCountyNodes[0].map((data) => {
     let averageResults = returnPMAverage(data.sensors);
 
     if (averageResults.length > 0) {
-      // get average if more than one sensor has data
+      // get average  + idif more than one sensor has data
+      const id = averageResults.map((item) => item.id)[0];
```

Comment:
Shouldn't this be just
```suggestion
      const id = data.id;
```
`data` being the node? Or what is `data`?